File: core/src/main/dc/targetman/mechanics/ScaleSystem.java
package dc.targetman.mechanics;

import com.badlogic.gdx.math.Vector2;
import dc.targetman.epf.parts.ScalePart;
import dclib.epf.Entity;
import dclib.epf.EntityManager;
import dclib.epf.EntitySystem;
import dclib.epf.parts.TransformPart;
import dclib.physics.Transform;
import dclib.util.Timer;

public final class ScaleSystem extends EntitySystem {

	public ScaleSystem(final EntityManager entityManager) {
		super(entityManager);
	}

	@Override
	protected final void update(final float delta, final Entity entity) {
		ScalePart scalePart = entity.tryGet(ScalePart.class);
		if (scalePart != null) {
			Timer scaleTimer = scalePart.getScaleTimer();
			scaleTimer.tick(delta);
			float scaleX = scalePart.getScaleX();
			Transform transform = entity.get(TransformPart.class).getTransform();
			transform.setScale(new Vector2(scaleX, transform.getScale().y));
		}
	}

}

File: core/src/main/dc/targetman/mechanics/ScaleSystem.kt
package dc.targetman.mechanics

import com.badlogic.gdx.math.Vector2
import dc.targetman.epf.parts.ScalePart
import dclib.epf.Entity
import dclib.epf.EntityManager
import dclib.epf.EntitySystem
import dclib.epf.parts.TransformPart

class ScaleSystem(entityManager: EntityManager) : EntitySystem(entityManager) {
    override fun update(delta: Float, entity: Entity) {
        val scalePart = entity.tryGet(ScalePart::class)
        if (scalePart != null) {
            scalePart.scaleTimer.tick(delta)
            val transform = entity[TransformPart::class].transform
            transform.scale = Vector2(scalePart.scaleX, transform.scale.y)
        }
    }
}
